src/script/lua_api/l_util.cpp

```diff
@@ -336,6 +336,50 @@ int ModApiUtil::l_mkdir(lua_State *L)
 	return 1;
 }
 
+// rmdir(path, recursive)
+int ModApiUtil::l_rmdir(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	const char *path = luaL_checkstring(L, 1);
+	CHECK_SECURE_PATH(L, path, true);
+
+	bool recursive = false;
+
+	if (!lua_isnil(L, 2))
+		recursive = lua_toboolean(L, 2);
+
+	if (recursive)
+		lua_pushboolean(L, fs::RecursiveDelete(path));
+	else
+		lua_pushboolean(L, fs::DeleteSingleFileOrEmptyDirectory(path));
+
+	return 1;
+}
+
+// cpdir(source, destination, remove_source)
+int ModApiUtil::l_cpdir(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	const char *source = luaL_checkstring(L, 1);
+	const char *destination = luaL_checkstring(L, 2);
+	CHECK_SECURE_PATH(L, source, false);
+	CHECK_SECURE_PATH(L, destination, true);
+
+	bool remove_source = false;
+
+	if (!lua_isnil(L, 3))
+		remove_source = lua_toboolean(L, 3);
+
+	bool retval = fs::CopyDir(source, destination);
+
+	if (retval && (remove_source)) {
+		CHECK_SECURE_PATH(L, source, true);
+		retval &= fs::RecursiveDelete(source);
+	}
```

Comment:
I don't think cpdir should support `remove_source`, you can trivially do this yourself in Lua with the provided APIs.

Reply:
I guess that fair, should I introduce a pure-Lua `mvdir` as rubenwardy suggested though? Without either it is trivial but still annoying having to call and pass the same path to both, possibly repeatedly.

Reply:
Implementing `mvdir` as copy & delete is inefficient, a potential `mvdir` should be a C++ implementation that tries to use [`rename(2)`](http://man7.org/linux/man-pages/man2/rename.2.html) wherever possible.

So in fact cpdir should absolutely not support `remove_source` to not encourage using it as a mvdir replacement.